_overviews/scala3-scaladoc/snippet-compiler.md

```diff
@@ -0,0 +1,226 @@
+
+```scala
+package scala.collection
+trait Snippet[A] { self: List[A] =>
+  slice(2,5) == drop(2).take(3)
+}
+```
+
+### Hiding code
+
+Despite having the context feature described above, sometimes an author needs to provide more elements to scope. A big block of imports and initializations of necessary classes can result in loss of readablity. On the other hand, we’ve read a lot of opinions that people would like to be able to see the whole code. For that case we introduced special syntax for snippets that hides certain fragments of code, but you can always expand the snippet in documentation.
+
+Example:
+
+```scala
+//{
+import scala.collection.immutable.List
+//}
+val intList: List[Int] = List(1, 2, 3)
+```
+
+![]({{ site.baseurl }}/resources/images/scala3/scaladoc/hiding-code.gif)
+
+### Snippet includes
+
+While writing code snippets, we often need a mechanism to reuse code from one snippet in another.
+
+Take a look at the following piece of documentation:
+
+![]({{ site.baseurl }}/resources/images/scala3/scaladoc/documentation-snippet.png)
+
+To successfully compile the last snippet, we need to have previously declared definitions in scope. For this scenario and probably many more we added a new feature: snippet includes. It allows you to reuse code from one snippet in another which results in less redundancy and better maintainability.
+
+To configure it, you need to add a `sc-name` argument directly in the snippet that you want to include:
+
+```` ```scala sc-name:<snippet-name> ````
+
+where `snippet-name` should be unique in file scope and cannot contain whitespaces and commas.
+
+Then, you can add a `sc-compile-with` argument directly in the snippet that should contain include:
+```` ```scala sc-compile-with:<snippet-name>(,<snippet-name>)+ ````
+
+where `snippet-name` is the name of snippet that should be included.
+
+After configuring this feature in our example, the code looks like this:
+
+![]({{ site.baseurl }}/resources/images/scala3/scaladoc/documentation-snippet2.png)
+
+And the output looks like this:
+
+![]({{ site.baseurl }}/resources/images/scala3/scaladoc/snippet-includes.png)
+
+You can specify more than one include. The order in which they are specified defines the order of inclusion. 
+
+**Warning**: you can only include snippets that are defined above the target snippet.
+
+## Advanced configuration
+
+Often turning on snippet valiation for all snippets is not enough as the usecases can be more sophisticated. We prepared our tool for such situations to allow user adjust it to his needs.
+
+### Available flags
+
+Snippet compiler exposes three flags that change its behavior:
+- `compile` - turns on snippet checking
+- `nocompile` - turns off snippet checking
+- `fail` - turns on snippet checking with assertion that snippets don't compile
+
+### Path-based settings
+
+Instead of setting one flag to all snippets in project, it can be set for certain path only. It can be done by adding `<path>=` prefix before flag.
```

Comment:
```suggestion
For more flexibility, instead of setting one flag to control all snippets in a project, it can be set for a certain path only, by adding `<path>=` prefix before flag. For example:
```